### PR TITLE
Rabbitmq-Native Metrics

### PR DESCRIPTION
**Pull request by aaron-brown**
@budjb: Here is the PR for the metrics. However, I have a couple of thoughts.

Given the addition of the Spring Application Events, I'm not sure that I like having the plugin do the [JMX Report Exporting](https://github.com/aaron-brown/grails-rabbitmq-native/blob/rabbitmq-native-metrics/rabbitmq-native/src/main/groovy/com/budjb/rabbitmq/connection/ConnectionContextImpl.groovy#L108-L116).

I modified the configuration of a connection to look like so:

```yaml
rabbitmq:
    connnections:
      - name: 'primary'
         # Other required fields...
         metrics:
             enabled: true
             jmx:
                 export: true
                 domain: 'com.aarocode.myapp.rabbitmq.connection.primary'
```

The reason I did so was because I did not want to make any assumptions about the ```domain``` field that might restrict the user. However, if JMX wasn't in the Plugin, I could still export it, and have a lot more control over that exporting, with the following:

```groovy
class RabbitMetricsJmxExportListener extends ApplicaitonListener<ConnectionManagerStartedEvent> {
    @Autowired
    ConnectionManager connectionManager

    @Override
    void onApplicationEvent(ConnectionManagerStartedEvent event) {
        connectionManager.contexts.each { context ->
            JmxReporter reporter = JmxReporter
                .forRegistry(metricRegistry)
                .inDomain("com.aarocode.myapp.rabbitmq.connection.${context.id}")
                .build()
            reporter.start()
        }
    }
}
```

Or, however else I wanted to build the **JmxReporter**.

So basically, while I've implemented it, I feel it might be better to remove it, and just have the plugin handle wiring the metrics to the **ConnectionFactory**, since that has to happen within the plugin, as that is not exposed in any way.

That also simplifies the configuration to just:

```yaml
rabbitmq:
    connnections:
      - name: 'primary'
         # Other required fields...
         metrics:
             enabled: true
        # Or even enableMetrics: true, to be a one-liner
```

Would like to hear your thoughts & opinions on that.

### CODE REVIEW

**Comment by aaron-brown** [comment]
@budjb: Updated PR to remove the JMX Exort logic, and changed the configuration key:

```yaml
rabbitmq:
    connections:
      - name: primary
        # Other necessary fields.
        enableMetrics: true
```

Metrics are disabled by default, which matches RabbitMQ's default behavior.
